### PR TITLE
Don't translate "add-on" string and use default translation for add-on name in editors emails

### PR DESCRIPTION
**Pull request by diox**
Fix mozilla/addons#133

### CODE REVIEW

File: src/olympia/editors/tests/test_review_scenarios.py [hunk 1]
```diff
@@ -26,7 +27,7 @@ def addon_with_files(db):
     By default the add-on is public, and the files are: beta, disabled,
     unreviewed, unreviewed.
     """
-    addon = Addon.objects.create()
+    addon = Addon.objects.create(name='My Addon', slug='my-addon')
```
Comment (eviljeff) [inline src/olympia/editors/tests/test_review_scenarios.py]: unrelated changes?

File: src/olympia/editors/templates/editors/emails/pending_to_public.ltxt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {% extends "editors/emails/base.ltxt" %}{% block content %}
-Your {{ addon_type }}, {{ name }} {{ number }}, has been fully reviewed and is now available for download in our gallery at {{ addon_url }}
+Your add-on, {{ name }} {{ number }}, has been fully reviewed and is now available for download in our gallery at {{ addon_url }}
```
Comment (eviljeff) [inline src/olympia/editors/templates/editors/emails/pending_to_public.ltxt]: Seems weird we're not using the `ADDON_TYPE` constants to say 'Extension', 'Dictionary', etc, but if we always say add-on now, then meh.

**Comment by eviljeff** [comment]
r+wc

File: src/olympia/editors/tests/test_review_scenarios.py (hunk 1 above)
Comment (diox) [inline src/olympia/editors/tests/test_review_scenarios.py]: No, it's related: those tests rely on this manually created add-on and then play with the review states using the helper class I modified. The add-on had no name (which should never happen in real life), and it was causing a failure

File: src/olympia/editors/templates/editors/emails/pending_to_public.ltxt (hunk 2 above)
Comment (diox) [inline src/olympia/editors/templates/editors/emails/pending_to_public.ltxt]: My reasoning exactly.
